kyaml/fn/runtime/runtimeutil/runtimeutil_test.go

```diff
@@ -1133,7 +1133,7 @@ metadata:
 `,
 			expectedFn: `
 container:
-    image: foo:v1.0.0`,
+  image: foo:v1.0.0`,
```

Comment:
The test cases in this file are showing us that Marshal was effectively using `SetIndent(4)` (the default), which did not match `Encode`. I think that was a mistake/bug, and we should change it since kyaml is still a zero-series release. But maybe in its own PR rather than as part of the go-yaml bump.

If you change Marshal to use `SetIndent(4)`, you'll also see what I believe is a problem with the yaml.v3 patch:

Expected:
```yaml
container:
    image: foo:v1.0.0
    mounts:
      - type: bind
        src: /mount/path
        dst: /local/
      - type: volume
        src: myvol
        dst: /local/
      - type: tmpfs
        dst: /local/
```

Actual:
```yaml
container:
    image: foo:v1.0.0
    mounts:
    - type: bind
      src: /mount/path
      dst: /local/
    - type: volume
      src: myvol
      dst: /local/
    - type: tmpfs
      dst: /local/
```

Reply:
>  I think that was a mistake/bug, and we should change it since kyaml is still a zero-series release.

Can we do that now, or is there any reason to wait?

Reply:
🤔 Good point, I don't see why we shouldn't do it now, ahead of the bump.

Reply:
I think this is resolved in go-yaml/yaml#753, but I want to do some more manual testing to make sure.